From: Sven Roederer <devel-sven@geroedel.de>
Date: Tue, 27 Apr 2021 00:47:19 +0200
Subject: ath79: Fix LAN-port for tplink_eap2x5-1port

The Lan port on TPLink EAP225 Outdoor v1 is unable to connect to an 100MBit/s
link and is intermittent on some 1GBit/s peers.

This patch fixes the issue by removing the phy-handle definition
and replacing it with a fixed-link definition

The WAN port is then able to connect to a modem via a link running at 100Mbs or 10Mbs

* ports b163bfbba59b62c05a4
* according to https://forum.openwrt.org/t/eap225-v1-lan-port-link-speed-100mbit-not-working/88060

diff --git a/target/linux/ath79/dts/qca9563_tplink_eap2x5-1port.dtsi b/target/linux/ath79/dts/qca9563_tplink_eap2x5-1port.dtsi
index 3978be75e663e57fc2c76893feb17004e3f92cde..4d99c234d9fbb3c6240e3eab5d0686452d9b9fe3 100644
--- a/target/linux/ath79/dts/qca9563_tplink_eap2x5-1port.dtsi
+++ b/target/linux/ath79/dts/qca9563_tplink_eap2x5-1port.dtsi
@@ -109,7 +109,6 @@
 &eth0 {
 	status = "okay";
 
-	phy-handle = <&phy4>;
 	phy-mode = "sgmii";
 	pll-data = <0x03000000 0x00000101 0x00001313>;
 
@@ -120,6 +119,11 @@
 	gmac-config {
 		device = <&gmac>;
 	};
+
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
 };
 
 &wmac {
